### DISCUSSION TITLE
CS1061	'AllureLifecycle' does not contain a definition for 'AddAttachment' and no accessible extension method 'AddAttachment' accepting a first argument of type 'AllureLifecycle' could be found (are you missing a using directive or an assembly reference?)	ATFramework (net6.0)

### DISCUSSION THREAD

**Discussion by Roman-Tretiakov** [allure-framework/allure2#2652] [Bug Reports] [answered]
I installed Allure.Net.Commons as alternative of Allure.Commons. And my old code doesn't work when I try to add screen shot attachment : 
 AllureLifecycle.Instance.Attachment("Screenshot", "image/png", screenshotPath); 

I have got following error: CS1061	'AllureLifecycle' does not contain a definition for 'AddAttachment'
I try to use AttachmentAsync method instead, but result is the same. Please help me.

**Comment by delatrie** [comment, marked as answer]
Please, use the `AllureApi` class instead:

```c#
AllureApi.AddAttachment("Screenshot", "image/png", screenshotPath)
```

It's in the `Allure.Net.Commons` namespace. You should add the following using statement at the top of the file:

```c#
using Allure.Net.Commons;
```

**Reply by Area51x** [reply]
Thanks, I had the same issue.

**Reply by mazharcoupa** [reply]
System.MissingMethodException : Method not found: 'Allure.Net.Commons.AllureLifecycle Allure.Net.Commons.AllureLifecycle.AddAttachment(System.String, System.String, Byte[], System.String)'.

  Stack Trace: 
AllureNUnitHelper.AddConsoleOutputAttachment()
AllureNUnitHelper.StopTestCase()
<>c__DisplayClass4_0.<AfterTest>b__0()
AllureNUnitAttribute.RunHookInRestoredAllureContext(ITest test, Action action)
AllureNUnitAttribute.AfterTest(ITest test)
<.ctor>b__1(TestExecutionContext _)
<>c__DisplayClass1_0.<Execute>b__1()
DelegatingTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)

I get this error when I use the above suggested code. How can I fix this

**Reply by delatrie** [reply]
@mazharcoupa 
Make sure Allure.NUnit and Allure.Net.Commons packages both have the same version
